packages/jest-message-util/src/index.ts

```diff
@@ -5,323 +5,294 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import * as path from 'path';
-import {codeFrameColumns} from '@babel/code-frame';
+import {
+  ErrorChainNode,
+  ErrorNode,
+  FrameNode,
+  getAbsoluteSitePath,
+  isInternalFrame as isNodeInternalFrame,
+  parseError,
+  parseErrors,
+  parseFrame,
+  printNode,
+} from '@stack-tools/node-tools';
 import chalk = require('chalk');
-import * as fs from 'graceful-fs';
-import micromatch = require('micromatch');
-import slash = require('slash');
-import StackUtils = require('stack-utils');
 import type {Config, TestResult} from '@jest/types';
 import {format as prettyFormat} from 'pretty-format';
 import type {Frame} from './types';
+import {
+  JestVisitor,
+  JestVisitorOptions,
+  getTopFrame as __getTopFrame,
+} from './visitor';
 
-export type {Frame} from './types';
+export type {Frame};
 
 type Path = Config.Path;
 
-// stack utils tries to create pretty stack by making paths relative.
-const stackUtils = new StackUtils({cwd: 'something which does not exist'});
-
-let nodeInternals: Array<RegExp> = [];
-
-try {
-  nodeInternals = StackUtils.nodeInternals();
-} catch {
-  // `StackUtils.nodeInternals()` fails in browsers. We don't need to remove
-  // node internals in the browser though, so no issue.
-}
-
 export type StackTraceConfig = Pick<
   Config.ProjectConfig,
   'rootDir' | 'testMatch'
 >;
 
 export type StackTraceOptions = {
-  noStackTrace: boolean;
+  noStackTrace?: boolean;
   noCodeFrame?: boolean;
+  indentHeader?: string;
+  indentFrames?: string;
 };
 
-const PATH_NODE_MODULES = `${path.sep}node_modules${path.sep}`;
-const PATH_JEST_PACKAGES = `${path.sep}jest${path.sep}packages${path.sep}`;
-
 // filter for noisy stack trace lines
-const JASMINE_IGNORE =
-  /^\s+at(?:(?:.jasmine\-)|\s+jasmine\.buildExpectationResult)/;
-const JEST_INTERNALS_IGNORE =
-  /^\s+at.*?jest(-.*?)?(\/|\\)(build|node_modules|packages)(\/|\\)/;
-const ANONYMOUS_FN_IGNORE = /^\s+at <anonymous>.*$/;
-const ANONYMOUS_PROMISE_IGNORE = /^\s+at (new )?Promise \(<anonymous>\).*$/;
-const ANONYMOUS_GENERATOR_IGNORE = /^\s+at Generator.next \(<anonymous>\).*$/;
-const NATIVE_NEXT_IGNORE = /^\s+at next \(native\).*$/;
-const TITLE_INDENT = '  ';
-const MESSAGE_INDENT = '    ';
-const STACK_INDENT = '      ';
```

Comment:
I removed these since their usage is inconsistent, i.e. message is not always indented four spaces and not everything indented four spaces is a message. There are many, many ways to figure out how many spaces there are in a string.